app/Repositories/ReceiptRepository/ReceiptRepository.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace App\Repositories\ReceiptRepository;
+
+use App\Models\Receipt;
+use App\Repositories\BaseRepository;
+use Illuminate\Support\Facades\DB;
+
+class ReceiptRepository extends BaseRepository implements ReceiptRepositoryInterface
+{
+
+    public function getModel()
+    {
+        return Receipt::class;
+    }
+
+    public function getCoinsPerMonth($years)
+    {
+        $coinMonth = [];
+        foreach ($years as $year) {
+            $coins = DB::table('receipts')
+                ->selectRaw('MONTH(created_at) as month, sum(value * quantity) as coin')
+                ->whereRaw('YEAR(created_at) = ' . $year->year)
+                ->groupBy(DB::raw('MONTH(created_at)'))
+                ->get();
```

Comment:
tương tự

Reply:
em tưởng câu này là n query thôi mà chị(n = số năm) : select MONTH(created_at) as month, sum(value * quantity) as coin from `receipts` where YEAR(created_at) = 2021 group by MONTH(created_at);

Reply:
em tưởng câu này là n query thôi mà chị( n = số năm) : `select MONTH(created_at) as month, sum(value * quantity) as coin from receipts where YEAR(created_at) = 2021 group by MONTH(created_at)`;

Reply:
độ phức tạp là O(n+12) nên nó là O(n), em có sửa lại để hạn chế for lồng, chị xem giúp em ạ (bow)